text/0000-additonal-oci-artifacts.md

```diff
@@ -0,0 +1,91 @@
+# Meta
+[meta]: #meta
+- Name: Additional OCI Artifacts from Buildpacks
+- Start Date: 2022-04-24
+- Author(s): agracey
+- Status: Draft <!-- Acceptable values: Draft, Approved, On Hold, Superseded -->
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: (put "N/A" unless this replaces an existing RFC, then link to that RFC)
+
+# Summary
+[summary]: #summary
+
+Allow buildpack authors to specify additional artifacts from their buildpacks to be uploaded to a registry along with the standard set of layers. 
+
+# Definitions
+[definitions]: #definitions
+
+Make a list of the definitions that may be useful for those reviewing. Include phrases and words that buildpack authors or other interested parties may not be familiar with.
+
+- WebAssembly (WASM) -- Portable binaries to be run with a WebAssembly virtual machine.
+- OCI Artifact -- Any blob of data that can be stored in an OCI registry
+
+# Motivation
+[motivation]: #motivation
+
+Cloud Native Buildpacks are awesome for building container images but come with a draw back that they can only produce a single output. This is limiting to platform builders because there are many cases where you might want to have multiple outputs from the same build process. 
+
+# What it is
+[what-it-is]: #what-it-is
+
+This change allows for additional flexibility in what buildpack authors and platforms can build on top of buildpacks. Two initial use cases are WebAssembly and Helm.
+
+For the first, a buildpack would be able to also generate a wasm bundle that can be used by a downstream platform (such as the Krustlet or runwasi). 
```

Comment:
Similarly, are there cases where you might want multiple different artifacts as an output from a single build? From the "How it Works" section, I gather this could be true (since there could be multiple layers with `additionalArtifacts` specified). Is it possible to give examples of those cases?

Reply:
My thinking for "also" was to reduce the size of the engineering effort needed and to allow for this feature to be layered on top of the existing functionality.  Would it be preferred to remove "also" and call out the option in an additional call out?

Reply:
I was thinking about it and it seems like making the standard image optional is a larger change than needed. I'd suggest waiting until we see adoption of this feature before adding complexity.